apps/src/templates/hintsDisplay.jsx

```diff
@@ -0,0 +1,57 @@
+module.exports = React.createClass({
+
+  propTypes: {
+    hintReviewTitle: React.PropTypes.string.isRequired,
+    getSeenHints: React.PropTypes.func.isRequired,
+    getUnseenHints: React.PropTypes.func.isRequired,
+    recordUserViewedHint: React.PropTypes.func.isRequired,
+    lightbulbSVG: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState: function () {
+    return {
+      seenHints: this.props.getSeenHints(),
+      unseenHints: this.props.getUnseenHints(),
+      showViewHintButton: true
+    };
+  },
+
+  viewHint: function () {
+    this.props.recordUserViewedHint(this.state.unseenHints[0]);
+    var newState = this.getInitialState();
```

Comment:
The alternative would of course be to pass in the lists of seen vs unseen hints as props and just modify them when the button is pressed to match the logic of the controller that's passing in those getters, but this seemed cleaner.